core/pkg/artifacts/multipart.go

```diff
@@ -0,0 +1,207 @@
+package artifacts
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math"
+	"os"
+	"runtime"
+	"time"
+
+	"github.com/wandb/wandb/core/internal/gql"
+	"github.com/wandb/wandb/core/internal/hashencode"
+	"github.com/wandb/wandb/core/internal/observability"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	S3MinMultiUploadSize = 2 << 30   // 2 GiB, the threshold we've chosen to switch to multipart
+	S3MaxMultiUploadSize = 5 << 40   // 5 TiB, maximum possible object size
+	S3DefaultChunkSize   = 100 << 20 // 100 MiB
+	S3MaxParts           = 10000
+)
+
+// createMultiPartRequest checks if the file size is large enough to use multipart upload.
+// If so, it computes the hash of each parts required for getting multipart upload url from server.
+// Otherwise, it returns nil, which is a valid input for the graphql CreateArtifactFileSpecInput.
+func createMultiPartRequest(logger *observability.CoreLogger, path string) ([]gql.UploadPartsInput, error) {
```

Comment:
```suggestion
func createMultiPartRequest(
	 logger *observability.CoreLogger,
	 path string,
) ([]gql.UploadPartsInput, error) {
```

unfortunately, `go fmt` does not do this automatically

Reply:
Done.